src/response/file.rs

```diff
@@ -73,15 +64,14 @@ impl File {
     /// Respond with the contents of this file.
     ///
     pub async fn serve(mut self) -> middleware::Result {
-        let mut file = fs::File::open(&self.path).await.map_err(wrap_io_error)?;
-        let metadata = file.metadata().await?;
+        let mut file = fs::File::open(&self.path).await.map_err(Error::from_io)?;
+        let metadata = file.metadata().await.map_err(Error::from_io)?;
 
-        let mut data = match metadata.len().try_into() {
-            Ok(capacity) => Vec::with_capacity(capacity),
-            Err(error) => return Err(Error::payload_too_large(error.into())),
-        };
+        // Allocate the exact capacity required to store the file in memory.
+        // This is
```

Comment:
I think I'm going to introduce a stream implementation here if the file is larger than `15MB`. I made the decision mid comment.